SimCalorimetry/HGCalSimProducers/src/HGCDigitizer.cc

```diff
@@ -99,42 +100,172 @@ namespace {
     const float minPackChargeLog = minCharge > 0.f ? std::log(minCharge) : -2;
     const float maxPackChargeLog = std::log(maxCharge);
-    constexpr uint16_t base = 1 << PHGCSimAccumulator::Data::sampleOffset;
+    constexpr uint16_t base = 1 << PreMixHGCSimAccumulator::SimHitCollection::sampleOffset;
 
     simResult.reserve(simData.size());
-    // mimicing the digitization
+
     for (const auto& id : validIds) {
       auto found = simData.find(id);
       if (found == simData.end())
         continue;
-      // store only non-zero
-      for (size_t iEn = 0; iEn < nEnergies; ++iEn) {
-        const auto& samples = found->second.hit_info[iEn];
-        for (size_t iSample = 0; iSample < hgc_digi::nSamples; ++iSample) {
-          if (samples[iSample] > minCharge) {
-            const auto packed = logintpack::pack16log(samples[iSample], minPackChargeLog, maxPackChargeLog, base);
-            simResult.emplace_back(id.rawId(), iEn, iSample, packed);
+
+      const hgc_digi::PUSimHitData& accCharge_accros_bxs = found->second.PUhit_info[0];
+      const hgc_digi::PUSimHitData& timing_accros_bxs = found->second.PUhit_info[1];
+      for (size_t iSample = 0; iSample < hgc_digi::nSamples; ++iSample) {
+        const std::vector<float>& accChrage_inthis_bx = accCharge_accros_bxs[iSample];
+        const std::vector<float>& timing_inthis_bx = timing_accros_bxs[iSample];
+        std::vector<unsigned short> vc, vt;
+        size_t nhits = accChrage_inthis_bx.size();
+
+        for (size_t ihit = 0; ihit < nhits; ++ihit) {
+          if (accChrage_inthis_bx[ihit] > minCharge) {
+            unsigned short c =
+                logintpack::pack16log(accChrage_inthis_bx[ihit], minPackChargeLog, maxPackChargeLog, base);
+            unsigned short t = logintpack::pack16log(timing_inthis_bx[ihit], minPackChargeLog, maxPackChargeLog, base);
+            vc.emplace_back(c);
+            vt.emplace_back(t);
           }
         }
+        simResult.emplace_back_timing(id.rawId(), iSample, vc, vt);
       }
     }
     simResult.shrink_to_fit();
   }
 
+  typedef std::vector<std::pair<float, float>>::iterator itr;
+  typedef std::map<uint32_t, std::vector<std::pair<float, float>>> IdHit_Map;
+  template <typename GEOM>
+  void loadSimHitAccumulator_forPreMix(hgc::HGCSimHitDataAccumulator& simData,
+                                       hgc::HGCPUSimHitDataAccumulator& PUSimData,
+                                       const GEOM* geom,
+                                       IdHit_Map& hitRefs_bx0,
+                                       const PreMixHGCSimAccumulator& simAccumulator,
+                                       const float minCharge,
+                                       const float maxCharge,
+                                       bool setIfZero,
+                                       const std::array<float, 3>& tdcForToAOnset,
+                                       const bool& minbiasFlag,
```

Comment:
```suggestion
                                       const bool minbiasFlag,
```
primitive types should be taken by value. Maybe `pileupFlag` or `isPileup` would be more clear.